packages/common/client-utils/src/test/mocha/layerCompat.spec.ts

```diff
@@ -0,0 +1,143 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "node:assert";
+
+import {
+	checkLayerCompatibility,
+	type ILayerCompatDetails,
+	type ILayerCompatSupportRequirements,
+	type LayerCompatCheckResult,
+} from "../../layerCompat.js";
+
+const pkgVersion = "1.0.0";
+
+describe("checkLayerCompatibility", () => {
+	it("should return not compatible when other layer doesn't support ILayerCompatDetails", () => {
+		const compatSupportRequirementsLayer1: ILayerCompatSupportRequirements = {
+			requiredFeatures: ["feature1", "feature2"],
+			minSupportedGeneration: 1,
+		};
+
+		const result: LayerCompatCheckResult = checkLayerCompatibility(
+			compatSupportRequirementsLayer1,
+			undefined /* compatDetailsLayer2 */,
+		);
+		const expectedResults: LayerCompatCheckResult = {
+			isCompatible: false,
+			isGenerationCompatible: false,
+			unsupportedFeatures: compatSupportRequirementsLayer1.requiredFeatures,
+		};
+		assert.deepStrictEqual(result, expectedResults, "Layers should be compatible");
+	});
+
+	it("should return compatible when both generation and features are compatible", () => {
+		const compatSupportRequirementsLayer1: ILayerCompatSupportRequirements = {
+			requiredFeatures: ["feature1", "feature2"],
+			minSupportedGeneration: 1,
+		};
+
+		const compatDetailsLayer2: ILayerCompatDetails = {
+			pkgVersion,
+			generation: 1,
+			supportedFeatures: new Set(["feature1", "feature2"]),
```

Comment:
Slightly more interesting test case

```suggestion
			supportedFeatures: new Set(["feature1", "feature2", "feature3"]),
```